### PR TITLE
feat: add ability to send erc20 tokens

### LINKED ISSUE #18413: Wallet: Add support for sending STT (Status Test Token)
On the select Asset page, if a user selects Status Test Token as the asset to send the confirm button does not respond.

STT token should work the same as any other token.

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #18413

### Summary

As a continuation of #18473 which adds support for sending ERC721 tokens, this PR enables sending ERC20 tokens, for example STT in Goerli, or SNT in mainnet.

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

- Login 
- Open wallet
- Select an account
- Select "Send" option
- Enter an address
- Select an ERC20 token in the Asset list such as STT
- Input any valid amount
- Check transaction confirmation screen is showing correct information
- Slide to execute transaction
- Verify "Sending..." page is shown and that the transaction is executed correctly (compare in blockchain explorer such as Etherscan)

status: ready

### CODE REVIEW

**Review by OmarBasem — APPROVED** [review]

File: src/status_im/contexts/wallet/send/events.cljs
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events.cljs:154]: is `eth-transfer` specific to eth? or should this be `token-transfer`?

File: src/status_im/contexts/wallet/send/events.cljs
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events.cljs:222]: perhaps "0x0" should be a named const? e.g one-in-hex

**Review by J-Son89 — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hey @briansztamfater! Thanx for the PR. One question: I see that this PR is based on [feat/send-collectible-events-4](https://github.com/status-im/status-mobile/tree/feat/send-collectible-events-4) branch, which has not been tested yet and in theory it might have changes/fixes in case the bugs are found. Maybe we should better combine these features into one PR to test them in one common branch? WDYT?

**Comment by pavloburykh** [comment]
> Hey @briansztamfater! Thanx for the PR. One question: I see that this PR is based on [feat/send-collectible-events-4](https://github.com/status-im/status-mobile/tree/feat/send-collectible-events-4) branch, which has not been tested yet and in theory it might have changes/fixes in case the bugs are found. Maybe we should better combine these features into one PR to test them in one common branch? WDYT?

In case if this is not a good idea, I guess we better test (https://github.com/status-im/status-mobile/tree/feat/send-collectible-events-4) first and after it will be merged we should base current PR brach against the develop.

**Comment by briansztamfater** [comment]
> Hey @briansztamfater! Thanx for the PR. One question: I see that this PR is based on [feat/send-collectible-events-4](https://github.com/status-im/status-mobile/tree/feat/send-collectible-events-4) branch, which has not been tested yet and in theory it might have changes/fixes in case the bugs are found. Maybe we should better combine these features into one PR to test them in one common branch? WDYT?

Hey @pavloburykh ! I have no strong opinion on this, I can adapt to whatever is better for you. The base PR contains major changes and as they are modifying the same part of the code I decided to use that branch as base, to avoid conflicts when merging it

File: src/status_im/contexts/wallet/send/events.cljs
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/events.cljs:154]: Yes, it is specific to ETH (or native token), for the rest of tokens we use `erc20-transfer?`

**Comment by pavloburykh** [comment]
> Hey @pavloburykh ! I have no strong opinion on this, I can adapt to whatever is better for you. The base PR contains major changes and as they are modifying the same part of the code I decided to use that branch as base, to avoid conflicts when merging it

We are currently kinda blocked with testing https://github.com/status-im/status-mobile/tree/feat/send-collectible-events-4 because we don't have a good way to get collectibles on testnet. Therefore we cannot test sending properly. 

So we can test current PR with sending erc20 tokens but as far as understand we cannot merge it until https://github.com/status-im/status-mobile/tree/feat/send-collectible-events-4 is tested.

**Comment by VolodLytvynenko** [comment]
@briansztamfater Thank you for PR. Here is a found issue:

## ISSUE 1: [Android] The buttons on sending page does not responds

### Steps:
1. Select any tokens to send
2. Slide to execute transaction
3. Tap 'done' or [x] buttons on 'Sending' page

### Actual result:
 'done' or [x] buttons does not close the 'sending' page


https://github.com/status-im/status-mobile/assets/52490791/6e13afa1-0b52-4d8d-b421-f60c1201c7bf



### Expected result:
 After 'done' or [x] buttons are tapped the 'sending' page should be closed

### ENV:
Real device: Pixel 7a, Android 14
 
### Logs:
[Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/13940991/Status-debug-logs.zip)

**Comment by J-Son89** [comment]
> @briansztamfater Thank you for PR. Here is a found issue:
> 
> 
> 
> ## ISSUE 1: [Android] The buttons on sending page does not responds
> 
> 
> 
> ### Steps:
> 
> 1. Select any tokens to send
> 
> 2. Slide to execute transaction
> 
> 3. Tap 'done' or [x] buttons on 'Sending' page
> 
> 
> 
> ### Actual result:
> 
>  'done' or [x] buttons does not close the 'sending' page
> 
> 
> 
> 
> 
> https://github.com/status-im/status-mobile/assets/52490791/6e13afa1-0b52-4d8d-b421-f60c1201c7bf
> 
> 
> 
> 
> 
> 
> 
> ### Expected result:
> 
>  After 'done' or [x] buttons are tapped the 'sending' page should be closed
> 
> 
> 
> ### ENV:
> 
> Real device: Pixel 7a, Android 14
> 
>  
> 
> ### Logs:
> 
> [Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/13940991/Status-debug-logs.zip)
> 
> 
> 
> 

That is a separate issue and I have addressed in a pr I opened today 👌

**Pull request merged**
